apps/serdes/Deserializer.cpp

```diff
@@ -0,0 +1,608 @@
+#include "Deserializer.h"
+#include <fstream>
+#include <iostream>
+
+std::string Deserializer::deserialize_string(const flatbuffers::String *str) {
```

Comment:
If `str` can't ever (legally) be null, we should pass it by const-ref rather than const-ptr.